native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -268,16 +269,16 @@ private static Object convertBXmlSequence(BXmlSequence xmlSequence, String attri
         if (newSequence.isEmpty()) {
             return null;
         }
-        return convertHeterogeneousSequence(attributePrefix, preserveNamespaces, newSequence, attributeManager, type,
-                uniqueKey);
+        return convertHeterogeneousSequence(attributePrefix, preserveNamespaces, newSequence, type,
+                uniqueKey, parentAttributeMap);
     }
 
     private static Object convertHeterogeneousSequence(String attributePrefix, boolean preserveNamespaces,
-                                                       List<BXml> sequence, AttributeManager attributeManager,
-                                                       Type type, String uniqueKey) {
+                                                       List<BXml> sequence, Type type, String uniqueKey,
```

Comment:
As discussed, shall we remove uniqueKey and pass the nested record type instead

Reply:
Removed [757c01d](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/422/commits/757c01d78a62f37b5ef669c58b576bf6b9764df9)